python/ct/crypto/in_memory_merkle_tree.py

```diff
@@ -0,0 +1,125 @@
+"""In-memory Merkle Tree
+
+Operates (and owns) an in-memory array of leaves which can be updated.
+Not particularly efficient.
+"""
+
+import math
+
+from ct.crypto import merkle
+
+def _down_to_power_of_two(n):
+    """Returns the power-of-2 closest to n"""
+    if n < 2:
+        raise ValueError("N should be >= 2: %d" % n)
+    log_n = math.log(n, 2)
+    p = int(log_n)
+    # If n is exactly power of 2 then 2**p would be n, decrease p by 1.
+    if p == log_n:
+        p = p - 1
```

Comment:
C6407: 19:_down_to_power_of_two: Use -= instead of - [g-no-augmented-assignment]